### DISCUSSION TITLE
Running locally, confused documentation

### DISCUSSION THREAD

**Discussion by abelncm** [nextcloud/all-in-one#1699] [Questions] [answered]
I managed to make it run with a nginx reverse proxy.

On application setup page, the domain name that i put, i get the message that the server is not reachable at port 433!

Also the 11000 port page is just a token/hash code, why would you proxy to this and not 8080! I don't get this.

Please help me so i can create a docker compose file with instructions to make nextcloud run on localhost.

Thank you

**Comment by szaimen** [comment]
Hi, see https://github.com/nextcloud/all-in-one#how-to-skip-the-domain-validation

**Reply by abelncm** [reply]
That worked. Another thing, I cleaned up nextcloud in my docker so i can start from a blank state. But it still gives me the login page, even though i removed everything from docker, even its volumes!

**Reply by szaimen** [reply]
Sorry, cannot explain this. Did you follow https://github.com/nextcloud/all-in-one#how-to-properly-reset-the-instance?

**Reply by abelncm** [reply]
I reinstalled my docker, I got the docker compose file for mastercontainer, i started with with the skip domain verification, but i'm using it with localhost. It passed and installed all the remaining containers, it gave me the admin credentials, now when i click open you nextcloud button it doesn't take me to the login page, it just goes to the same main page! Is there a specific url path it should go to?

Also I didn't configure a proxy, nothing. I'm trying to see if with minimal configuration i could make this work.

**Comment by szaimen** [comment]
All right! However AIO cannot work without a domain. For a local instance, see this documenation: https://github.com/nextcloud/all-in-one/blob/main/local-instance.md

**Reply by abelncm** [reply]
Ok i'm following the docs, in general I know what to do. Only thing i want you to help me with is this. There's this container apache image that the docker-compose file pulled, nextcloud-aio-apache, and it's running on port 443. This won't let Caddy use that port.
Should I stop the apache container that's running on 443 so I can have Caddy for reverse proxy?

**Reply by szaimen** [reply]
You need to use a different port than port 443 by setting the variable APACHE_PORT to a different value. Restarting the containers should from then on use the chosen port

**Reply by abelncm** [reply]
Ok. Almost done. Last thing... I followed the instructions, the 1 method, the recommended one. Seems like for the first method, since the ACME challenge is mentioned, I should be getting the ACME challenge message, but I'm getting that my challenge failed for my domain.
Is there a way i can disable this? Or should I also follow method 2?

**Comment by szaimen** [comment]
The ACME challenge is used in method 1 to get a valid certificate. If thst fails for you because some of the requirements are not fullfilled, you can try method 2.

**Reply by abelncm** [reply]
I managed to work around this, I will share my docker compose file soon after fixing this one issue with this all in one nextcloud.
You know the first time I cleaned up my docker to start from scratch, the AIO app was still running? Ya that's happening again.
I removed everything, AIO still running, even after restarting docker service or my computer. This is the following process that's using the 8080 port:

tcp        0      0 0.0.0.0:8080            0.0.0.0:*               LISTEN      2449/docker-proxy   
tcp6       0      0 :::8080                 :::*                    LISTEN      2460/docker-proxy

The previous time I had to reinstall docker. But this time I need to know why is the APP up even though nothing is in docker anymore.

**Comment by szaimen** [comment]
Hm, did you stop all containers before trying to remove them? See https://github.com/nextcloud/all-in-one#how-to-properly-reset-the-instance

**Reply by abelncm** [reply]
I can't say for sure, but i think i did.

**Reply by szaimen** [reply]
Hm... What is the outout of `sudo docker ps -a`?

**Reply by abelncm** [reply]
Daamnn!!! I was only doing `docker ps -a` and it was showing me nothing!!
With sudo I can see that everything is there!!
Thank you. I'll fix it

**Comment by abelncm** [comment, marked as answer]
Hi, I hope this is helpful to you. It's a mininal config. No need for a DNS server, nor creating a certificate since Caddy handles that automatically.

https://github.com/abelncm/nextcloud-aio-local-docker-compose

You can test it if you want.

**Reply by szaimen** [reply]
Interesting! 

Your guide looks like https://github.com/nextcloud/all-in-one/discussions/588#discussioncomment-3450343. 

I'll mark your post as the answer of this thread 👍

**Reply by abelncm** [reply]
Yes, also just putting the domain in Caddyfile, for example: `local.docs` caddy generates a Lets Encrypt certificate and treats it as a HTTPS domain name. There's no need to do generate it manually and then mount the volume certs. Caddy is really handy in that regard. Even simpler than nginx.

**Reply by szaimen** [reply]
Yes, caddy is amazing, I know! :)

**Reply by abelncm** [reply]
One last thing. I know this docker compose file is supposed to have improved performance over the original Nextcloud installation, with redis, fulltext search, etc.
But the page rendering is slow, specially when jumping from one menu to another, the menu desapears (i know it's not a Single Page Application). So I added gzip encoding and header cache in the reverse proxy, but I feel like i need to improve it even further.
Can you point me in a direction so i can configure it to have shorter page load times?

If i managed to improve it i could then share it in my github.

**Reply by szaimen** [reply]
There is this: https://github.com/nextcloud/all-in-one/discussions/1335
